app/models/manageiq/providers/kubernetes/container_manager/scanning/job.rb

```diff
@@ -316,10 +331,27 @@ def pod_definition
           }
         ]
       }
-    )
+    }
+
+    inspector_admin_secret_name = inspector_admin_secret
+    add_secret_to_pod_def(pod_def, inspector_admin_secret_name) unless inspector_admin_secret_name.blank?
+
+    Kubeclient::Pod.new(pod_def)
+  end
+
+  def add_secret_to_pod_def(pod_def, inspector_admin_secret_name)
+    pod_def[:spec][:containers][0][:command].append("--dockercfg=" + INSPECTOR_ADMIN_SECRET_PATH +
+                                                    inspector_admin_secret_name + "/.dockercfg")
+    pod_def[:spec][:containers][0][:volumeMounts].append(
+      :name      => "inspector-admin-secret",
+      :mountPath => INSPECTOR_ADMIN_SECRET_PATH + inspector_admin_secret_name,
+      :readOnly  => true)
+    pod_def[:spec][:volumes].append(
+      :name   => "inspector-admin-secret",
+      :secret => {:secretName => inspector_admin_secret_name})
```

Comment:
`add_secret_to_pod_def` seems a little ugly. We're sure about the existence of all those indexes (e.g. `[:spec][:containers][0][:command]`), so I don't think we need to handle missing arrays/keys, etc. Anyway this is not looking nice. Not sure if @blomquisg or @Fryguy knows how to make this better. No need to block on this though.